src/sentry/releasehealth/metrics.py

```diff
@@ -340,3 +346,81 @@ def _count_users(total: bool, referrer: str) -> Dict[Any, int]:
             rv[project_id, release] = adoption
 
         return rv
+
+    def check_has_health_data(
+        self, projects_list: Sequence[ProjectOrRelease]
+    ) -> Set[ProjectOrRelease]:
+        now = datetime.now(pytz.utc)
+        start = now - timedelta(days=3)
+
+        projects_list = list(projects_list)
+
+        if len(projects_list) == 0:
+            return set()
+
+        includes_releases = isinstance(projects_list[0], tuple)
+
+        if includes_releases:
+            project_ids: List[ProjectId] = [x[0] for x in projects_list]  # type: ignore
+        else:
+            project_ids = projects_list  # type: ignore
+
+        org_id = self._get_org_id(project_ids)
+
+        where_clause = [
+            Condition(Column("org_id"), Op.EQ, org_id),
+            Condition(Column("project_id"), Op.IN, project_ids),
+            Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+            Condition(Column("timestamp"), Op.GTE, start),
+            Condition(Column("timestamp"), Op.LT, now),
+        ]
+
+        if includes_releases:
+            releases = [x[1] for x in projects_list]  # type: ignore
+            release_column_name = tag_key(org_id, "release")
+            releases_ids = [
+                release_id
+                for release_id in [try_get_tag_value(org_id, release) for release in releases]
+                if release_id is not None
+            ]
+            where_clause.append(Condition(Column(release_column_name), Op.IN, releases_ids))
+            column_names = ["project_id", release_column_name]
+
+            # def extract_raw_info(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
+            #     return row["project_id"], reverse_tag_value(org_id, row.get(release_column_name))  # type: ignore
+
+        else:
+            column_names = ["project_id"]
+
+            # def extract_raw_info(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
+            #     return row["project_id"]  # type: ignore
+
+        def extract_raw_info_func(
+            include_releases: bool,
+        ) -> Callable[[Mapping[str, Union[int, str]]], ProjectOrRelease]:
+            def f(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
+                if include_releases:
+                    return row["project_id"], reverse_tag_value(org_id, row.get(release_column_name))  # type: ignore
+                else:
+                    return row["project_id"]  # type: ignore
+
+            return f
+
+        extract_raw_info = extract_raw_info_func(includes_releases)
+
+        query_cols = [Column(column_name) for column_name in column_names]
+        group_by_clause = query_cols
```

Comment:
What is the point of defining `group_by_clause` here?

Reply:
some people do it to signal intent (`Query(select=query_cols, groupby=query_cols)` looks like a copypaste error) but I wouldn't have done it either